### PR TITLE
finished 4 tasks from ubunta

### PR DESCRIPTION
**Pull request by YaroslavBelinskyi**

### CODE REVIEW

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// 1 task
+const exampleArray = [10, -10, 10, -10, 10];
+const resultArray = [];
+function sumArray(arr) {
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: side effects? really?

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    arr.reduce((previousSum, currentNumber) => {
+        const value = previousSum + currentNumber;
+        resultArray.push(value);
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: pushing to array in global scope? not nice at all...

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    const objKeys = Object.keys(obj);
+    array.forEach((element) => {
+        for (let i = 0; i < objKeys.length; i++) {
+            if (objKeys[i] === element) {
+                const key = objKeys[i];
+                obj[key] += 1;
+            }
+        }
+    });
+    console.log(obj);
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: too many actions... you could do it in a single `reduce()` statement

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+}
+repeatsInObj(classes);
+// 3 task, I know this the easiest way to copy everything :D
+const objForCopy = { b: 'f', d: { e: 'f' } };
+function innerCopy(object) {
+    const objParsed = JSON.stringify(object);
+    const objCopy = JSON.parse(objParsed);
+    return objCopy;
+}
+const objInnerCopy = innerCopy(objForCopy);
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: nope, needs to be solved in different way, not using JSON methods

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        if (peopleArray[i].friends != null) {
+            if (peopleArray[i].friends.indexOf(index) >= 0) {
+                peopleWithFriends.push(peopleArray[i]);
+            }
+        }
+    }
+    if (peopleWithFriends.length < 1) {
+        peopleWithFriends = null;
+    }
+    console.log(peopleWithFriends);
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: to many `if` statements... please, think how to make your code more readable...

1. find user entry you need
2. loop through user entry `friends` array

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js (hunk 1 above)
Comment (YaroslavBelinskyi) [reply homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: fixed

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js (hunk 2 above)
Comment (YaroslavBelinskyi) [reply homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: fixed

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js (hunk 3 above)
Comment (YaroslavBelinskyi) [reply homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: fixed

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js (hunk 4 above)
Comment (YaroslavBelinskyi) [reply homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: done

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js (hunk 5 above)
Comment (YaroslavBelinskyi) [reply homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: a little bit improve my variant

**Comment by YaroslavBelinskyi** [comment]
please help me with git, i tried to resolve it by myself but nothing works.  it uploads other users files and started to give me errors on them. Now i cannot update my fork repo.

**Comment by YaroslavBelinskyi** [comment]
looks like I fixed it

**Comment by YaroslavBelinskyi** [comment]
finally fixed :D

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    console.log(obj);
+}
+repeatsInObj(classes);
+// 3 task
+const objForCopy = { b: 'f', d: { e: 'f' } };
+function objCopy(obj) {
+    if (typeof obj === 'object') {
+        return Object.keys(obj)
+            .map(k => ({ [k]: objCopy(obj[k]) }))
+            .reduce((a, c) => Object.assign(a, c), {});
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: idea is right, but make it more readable, you just need to look at each object property and do action according this property data type

keep it simple please

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    let peopleWithFriends = [];
+    for (let i = 0; i < peopleArray.length; i++) {
+        if (peopleArray[i].friends != null && peopleArray[i].friends.indexOf(index) >= 0) {
+            peopleWithFriends.push(peopleArray[i]);
+        }
+    }
+    if (peopleWithFriends.length < 1) {
+        peopleWithFriends = null;
+    }
+    console.log(peopleWithFriends);
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: again, to many unneeded operations, this solution may work but other developers will spend tons of their precious time to read it... please keep it simple as much as you can

1. find a user by index and return his friends ids
2. then simply filter initial array

Comment (YaroslavBelinskyi) [reply homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js]: Ii thought my was more simpler ;(

**Pull request merged**